nimble/host/src/ble_att_svr.c

```diff
@@ -636,29 +638,43 @@ ble_att_svr_tx_rsp(uint16_t conn_handle, uint16_t cid, int hs_status, struct os_
 
     if (do_tx) {
         if (hs_status == 0) {
+            conn = ble_hs_conn_find(conn_handle);
+            if (conn == NULL) {
+                hs_status = BLE_HS_ENOTCONN;
+                goto done;
+            }
+
+            chan = ble_hs_conn_chan_find_by_scid(conn, cid);
+            if (chan == NULL) {
+                hs_status = BLE_HS_ENOENT;
+                goto done;
+            }
+
+            ble_att_truncate_to_mtu(chan, om);
             hs_status = ble_att_tx(conn_handle, cid, om);
             om = NULL;
             if (hs_status) {
                 err_status = BLE_ATT_ERR_UNLIKELY;
             }
         }
 
-        if (hs_status != 0) {
-            STATS_INC(ble_att_stats, error_rsp_tx);
+done:
+    if (hs_status != 0) {
```

Comment:
nitpick: indentation (one level deeper, as it was), from here to end

Reply:
Should be ok now